go/vt/vtctl/workflow/stream_migrator.go

```diff
@@ -156,7 +155,8 @@ func (sm *StreamMigrator) CancelMigration(ctx context.Context) {
 	_ = sm.deleteTargetStreams(ctx)
 
 	err := sm.ts.ForAllSources(func(source *MigrationSource) error {
-		query := fmt.Sprintf("update _vt.vreplication set state='Running', stop_pos=null, message='' where db_name=%s and workflow != %s", encodeString(source.GetPrimary().DbName()), encodeString(sm.ts.ReverseWorkflowName()))
+		query := fmt.Sprintf("update _vt.vreplication set state='Running', stop_pos=null, message='' where db_name=%s and workflow=%s",
+			encodeString(source.GetPrimary().DbName()), encodeString(sm.ts.WorkflowName()))
 		_, err := sm.ts.VReplicationExec(ctx, source.GetPrimary().Alias, query)
 		return err
```

Comment:
can you explain the difference here.
Looks like earlier is used to bulk update as `workflow !=` would match more rows than the new change `workflow =`. I am not sure the implications here but just stating it out.

Reply:
In general the difference is what is described in the PR. 🙂 In this specific case, it's taking an unsafe query which was making an assumption about the data — namely that there's only one workflow in the keyspace and thus only records for the forward and reverse variants of it — and making it a safe query which only updates records for the specific workflow (and forward/reverse variant of it) that it should be.